### PR TITLE
HW 2 Submission

### PR DESCRIPTION
**Pull request by sauravpdx**

### CODE REVIEW

**Comment by sauravpdx** [comment]
**Code changes I made**
Major changes include counting the appearances of the words, changing color based on seconds input, and registration form 

**Why I made those changes**
This was a part of my HW2 assignment following the instructions

**What feedback am I looking for**
I am open to all kinds of feedback, primarily if I missed some use cases or style guides and also some suggestions to improve  and make the code more readable

File: hw2/01-flexbox.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+15. order:-3
+16. align-self: flex-end;
+17. align-self: flex-end; order: 1;
+18. order:1;align-self :flex-end;
+19. flex-wrap: wrap;
+20. flex-wrap: wrap;flex-direction: column
+21. align-content: flex-start;
+22. align-content: flex-end;
+23. flex-direction: column-reverse;align-content: center;
+24. can't solve
```
Comment (shrutidaundkar) [inline hw2/01-flexbox.md]: Try using reverse instead of the general flow.

File: hw2/02-cssgrid.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+12. grid-row-start : 3
+13. grid-row: 3/6
+14. grid-column: 2;grid-row: 5;
+15. can't solve
+16. grid-area: 1 / 2 / 4 / 6;
+17. can't solve
+18. order : 1
+19. order : -1
+20. grid-template-columns: 50% 50%;
+21.
```
Comment (shrutidaundkar) [inline hw2/02-cssgrid.md]: Look into the repeat() method and its desired parameters(mentioned in the hint).

File: hw2/03-count.js
Comment (shrutidaundkar) [inline hw2/03-count.js:1]: You can also consider using string and array methods like split(), join, map(), etc.

File: hw2/04-background.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    if (!isStarted) {
+      start();
+      toggleButton.textContent = "Stop";
+      toggleButton.classList.remove("btn-primary");
+      toggleButton.classList.add("btn-danger");
+    } else {
+      stop();
+      toggleButton.textContent = "Start";
+      toggleButton.classList.remove("btn-danger");
+      toggleButton.classList.add("btn-primary");
```
Comment (shrutidaundkar) [inline hw2/04-background.js]: You can try removing code redundancy by creating a different function.

**Review by shrutidaundkar — APPROVED** [review]
Good work!

File: hw2/01-flexbox.md [hunk 4]
```diff
@@ -2,6 +2,25 @@
...
```
Comment (mahathi0412) [inline hw2/01-flexbox.md]: Try using reverse for flex-flow and justify the content to the center and align the spaces.

File: hw2/04-background.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  let backgroundColorInterval;
+  let isStarted = false;
+
+  // reference: https://www.rustcodeweb.com/2021/07/how-to-generate-random-rgb-color-using-javascript.html
+  const randomColor = () => {
+    const x = Math.floor(Math.random() * 256);
+    const y = Math.floor(Math.random() * 256);
+    const z = Math.floor(Math.random() * 256);
+    const a = 0.2;
+    const RGBColor = "rgb(" + x + "," + y + "," + z + "," + a + ")";
```
Comment (A-Basirat) [inline hw2/04-background.js]: According to the Airbnb Style, we should use template literals for string concatenation. For example, when generating the RGB color string, use template literals:

`const RGBColor = `rgb(${x},${y},${z},${a})`;`

**Review by A-Basirat — COMMENTED** [review]
Your code looks good and seems to be functional. Great job

File: hw2/04-background.js (hunk 5 above)
Comment (sauravpdx) [reply hw2/04-background.js]: Thanks for pointing out Fixed it!

File: hw2/01-flexbox.md (hunk 1 above)
Comment (sauravpdx) [reply hw2/01-flexbox.md]: thanks it helps

File: hw2/03-count.js
Comment (sauravpdx) [reply hw2/03-count.js:1]: Thnaks for the suggestions shruti!

File: hw2/01-flexbox.md (hunk 4 above)
Comment (sauravpdx) [reply hw2/01-flexbox.md]: Thanks it helps updated the code

File: hw2/04-background.js (hunk 3 above)
Comment (sauravpdx) [reply hw2/04-background.js]: Thanks for the suggestions! Improved

File: hw2/02-cssgrid.md (hunk 2 above)
Comment (sauravpdx) [reply hw2/02-cssgrid.md]: Thanks it helps!

**Pull request merged**